edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java

```diff
@@ -132,6 +132,12 @@ private boolean doAction(LogInvocation logInvocation) {
 			}
 		}
 
+		// DOI not in intervals
+		if ((this.useLogCategory && (logInvocation.getDegreeOfInterestValue() > this.boundary.get(4)))) {
```

Comment:
This seems way too specific. Shouldn't this be just `logInvocation.getDegreeOfInterestValue() > this.boundary.get(this.boundary.size() - 1)`?

Reply:
Because if `useLogCategory` is turned on, the boundary list only has four elements.

Reply:
The `results.csv` after changes:

sequence | subject | repo URL | input logging statements | passing logging statements | failures | transformed logging statements | average Java lines added | average Java lines removed | log level not lowered in a catch block | log level not transformed due to if condition | use log category (SEVERE/WARNING/CONFIG) | use log category (CONFIG) | not lower log levels of logs inside of catch   blocks | time (s)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1555372844810 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 79 | 238 | 75 | 0 | 2 | FALSE | FALSE | FALSE | 19.531
1555372865188 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 75 | 0 | 0 | 4 | 2 | FALSE | FALSE | TRUE | 11.133
1555372876703 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 78 | 0 | 0 | 0 | 2 | FALSE | TRUE | FALSE | 12.575
1555372889666 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 74 | 0 | 0 | 4 | 2 | FALSE | TRUE | TRUE | 12.246
1555372902178 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 28 | 0 | 0 | 0 | 2 | TRUE | FALSE | FALSE | 11.91
1555372914135 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 27 | 0 | 0 | 4 | 2 | TRUE | FALSE | TRUE | 9.781

The results.csv before the changes:


sequence | subject | repo URL | input logging statements | passing logging statements | failures | transformed logging statements | average Java lines added | average Java lines removed | log level not lowered in a catch block | log level not transformed due to if condition | use log category (SEVERE/WARNING/CONFIG) | use log category (CONFIG) | not lower log levels of logs inside of catch   blocks | time (s)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1555357574413 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 79 | 238 | 75 | 0 | 2 | FALSE | FALSE | FALSE | 20.095
1555357595371 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 75 | 0 | 0 | 4 | 2 | FALSE | FALSE | TRUE | 11.66
1555357607451 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 78 | 0 | 0 | 0 | 2 | FALSE | TRUE | FALSE | 9.717
1555357617430 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 74 | 0 | 0 | 4 | 2 | FALSE | TRUE | TRUE | 9.458
1555357627124 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 34 | 0 | 0 | 0 | 2 | TRUE | FALSE | FALSE | 9.346
1555357636526 | deviation-upload | https://github.com/ponder-lab/deviation-upload.git | 91 | 91 | 0 | 33 | 0 | 0 | 4 | 2 | TRUE | FALSE | TRUE | 9.624

All files are attached on Asana.